.travis.yml

```diff
@@ -1,31 +1,65 @@
+env:
+  global:
+    # include $HOME/.local/bin for `aws`
+    - PATH=$HOME/.local/bin:$PATH
 language: shell
 notifications:
   email:
     on_success: never
 stages:
-- tests
+- unit_tests
+- build
+- e2e_tests
 - deployment
+before_install:
+  # set up awscli packages
+  - pip install --user awscli
+  # set up shared storage for build artifacts
+  - mkdir -p ~/build-stages-shared-storage/$TRAVIS_BUILD_NUMBER
+  - aws s3 sync s3://aws-log-forwader-travis-build-shared-storage/$TRAVIS_BUILD_NUMBER ~/build-stages-shared-storage/$TRAVIS_BUILD_NUMBER
 jobs:
   include:
-    - stage: tests
-      name: Tests & Linting
+    - stage: unit_tests
+      name: Execute unit tests & linting
       language: shell
       services: docker
       script:
         - docker build -f pipeline/Dockerfile -t lambda_image .
         - docker run --rm lambda_image bash -c 'python3 -m pylint --rcfile=pipeline/pylint.cfg $(find src/ -name "*.py")'
         - docker run --rm lambda_image bash -c 'cd src && python3 -m pytest -v ../tests/unit'
+    - stage: build
+      name: Build AWS Lambda package
+      #if: tag =~ /^release.*$/
+      language: shell
+      services: docker
+      script:
+        - docker run --rm -v `pwd`:/var/task amazon/aws-sam-cli-build-image-python3.8 bash -c './build-release-package.sh'
+        - cp -p dynatrace-aws-log-forwarder.zip ~/build-stages-shared-storage/$TRAVIS_BUILD_NUMBER
+        - cp -p dynatrace-aws-log-forwarder-lambda.zip ~/build-stages-shared-storage/$TRAVIS_BUILD_NUMBER
+    - stage: e2e_tests
+      name: Execute e2e tests
+      if: branch = master OR type = pull_request
+      language: shell
+      services: docker
+      script:
+        # deploy log forwarder resources reqiured for e2e to AWS
+        - cp -r ~/build-stages-shared-storage/$TRAVIS_BUILD_NUMBER/dynatrace-aws-log-forwarder-lambda.zip .
+        - ./dynatrace-aws-logs.sh deploy --target-url $DYNATRACE_ENV_URL --target-api-token $DYNATRACE_API_KEY --require-valid-certificate $VERIFY_SSL --stack-name aws-log-forwarder-e2e-build-$TRAVIS_BUILD_NUMBER

```

Comment:
Is $TRAVIS_BUILD_NUMBER unique globally?

Reply:
Yes if we are thinking in terms of repo scope.

Reply:
Can you give an example of its value? 
I would add a timestamp to the name, just as we do in integration test for AWS Metric Streaming. We did this to ensure unique names. Seems not really needed here but could add this just to be sure. But it's up to you

Reply:
The value is simply the build number i.e. 171. It was the easiest unique number to base ID on because it is stored as env variable by Travis.
If we have a common best practice in generating unique IDs, I'm more than happy to comply;)

Reply:
Nah, we don't have any common practices. Seems good. 
In previous tests we were using jenkins and there build number was not unique, each branch has its own numbering and conflicts could have come up